llvm/lib/CodeGen/MachineSink.cpp

```diff
@@ -1748,12 +1752,12 @@ bool MachineSinking::aggressivelySinkIntoCycle(
       }
       SinkBlock->insert(SinkBlock->SkipPHIsAndLabels(SinkBlock->begin()),
                         NewMI);
-      SunkInstrs[MapEntry] = NewMI;
+      SunkInstrs.insert({MapEntry, NewMI});
     }
 
     // Conservatively clear any kill flags on uses of sunk instruction
-    for (MachineOperand &MO : NewMI->operands()) {
-      if (MO.isReg() && MO.readsReg())
+    for (MachineOperand &MO : NewMI->all_uses()) {
+      if (MO.isReg())
         RegsToClearKillFlags.insert(MO.getReg());
```

Comment:
Might need to check or assert isUse, I can never remember the intricacies of all the operand iterators